### PR TITLE
Add slice op.

### PR DESCRIPTION
**Pull request by wanghaoshuang**
https://github.com/onnx/onnx/blob/master/docs/Operators.md#slice

### CODE REVIEW

File: paddle/fluid/operators/slice_op.cc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+Produces a slice of the input tensor along multiple axes. Similar to numpy:
+https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html
+Slices uses axes, starts and ends attributes to specify the start and end dimension
+for each axis in the list of axes, it uses this information to slice the input data tensor.
+If a negative value is passed for any of the start or end indices, it represent number of
+elements before the end of that dimension. If the value passed to start or end is larger
+than the n (the number of elements in this dimension), it represents n. For slicing to the
+end of a dimension with unknown size, it is recommended to pass in INT_MAX. If axes are omitted,
+they are set to [0, ..., ndim-1].
```
Comment (qingqing01) [inline paddle/fluid/operators/slice_op.cc]: Each line should not be more than 80 characters. Please fix the lines above.

File: paddle/fluid/operators/slice_op.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+#include <vector>
+#include "paddle/fluid/framework/op_registry.h"
+
+namespace paddle {
+namespace operators {
+
+using Tensor = framework::Tensor;
+template <typename T, size_t D, int MajorType = Eigen::RowMajor,
+          typename IndexType = Eigen::DenseIndex>
+using EigenTensor = framework::EigenTensor<T, D, MajorType, IndexType>;
```
Comment (qingqing01) [inline paddle/fluid/operators/slice_op.h]: Do not use `using` in the header file.

**Comment by tonyyang-svail** [comment]
Will there be a slice_grad_op?

**Comment by wanghaoshuang** [comment]
@tonyyang-svail  `slice op` is now used to slice tensor's shape whose gradient is not computed. But maybe it is necessary for some other scene.

**Pull request merged**